feat: Prepare PHP Release Docs (#2242)
getsentry/sentry-docs: src/platforms/php/common/performance.mdx

You need to add a [handler](http://docs.guzzlephp.org/en/stable/handlers-and-middleware.html#handlers) to your Guzzle Client. We provide one within the SDK which can be used like this:

```php
use Sentry\Tracing\GuzzleTracingMiddleware;
...

$stack = HandlerStack::create();

// This is the important line
$stack->push(GuzzleTracingMiddleware::trace());
// ----

$client = new Client([
    'base_uri' => 'http://httpbin.org',
    'timeout'  => 2.0,
    'handler' => $stack
]);
$client->request('GET', '/get');
```

After adding this handler, you will receive spans for every request done with this client.

### Manual Instrumentation

To manually instrument certain regions of your code, you can create a transaction to capture them.

```php
$transactionContext = new TransactionContext();
$transactionContext->setName('External Call');
$transactionContext->setOp('http.caller');
$transaction = \Sentry\startTransaction($transactionContext);

$spanContext = new SpanContext();
$spanContext->setOp('functionX');
$span1 = $transaction->startChild($spanContext);

// Calling functionX
$this->functionX();
$span1->finish();

$transaction->finish();
```

### Retrieving a Transaction

> In cases where you want to attach Spans to an already ongoing Transaction you can use `SentrySdk::getCurrentHub()->getTransaction()`. This function will return a `Transaction` in case there is a running Transaction on the scope, otherwise it returns `null`.

Reviewer: ```suggestion
If you want to attach Spans to an already ongoing Transaction, you can use `SentrySdk::getCurrentHub()->getTransaction()`. This function will return a `Transaction` if there is a running Transaction on the scope, otherwise it returns `null`.
```

```php
$transaction = SentrySdk::getCurrentHub()->getTransaction();

if ($transaction instanceof Transaction) {
    $transaction->setName($routeName);
    $transaction->setData([
        'action' => $route->getActionName(),
        'name' => $route->getName()
    ]);
}
```

> This can also be used if you want to filter for specifc conditions to not send a `Transaction` here is another example how to do it:

Reviewer: ```suggestion
You can also use this to filter for specific conditions when you don't want to send a `Transaction`. This example illustrates how:
```

```php
$transaction = SentrySdk::getCurrentHub()->getTransaction();
            
if ($transaction instanceof Transaction) {
    // $transaction->setSampled(false); // __DONT__ SEND TRANSACTION
    // $transaction->setSampled(true); // __DO__ SEND TRANSACTION  
}
```

> ### Using Relay as an Agent to improve your response time

Reviewer: ```suggestion
### Improve Response Time
```

Using the performance capabilities of PHP has some impact on your response time (depending on the `traces_sample_rate` you configured).

> Because of the nature of PHP, requests in most cases, are sent in the same process as you serve your users response.

Reviewer: ```suggestion
Because of the nature of PHP, requests in most cases are sent in the same process as you serve your users' response.
```

> Simply explained this means that the time it takes to send to request to Sentry, this time will be added on top of your servers response time. To mitigate that and almost reduce this time to zero, you can use a Relay running locally on the same machine or local network that acts as a Proxy / Agent. So instead of your PHP process sending things to Sentry, the PHP process sends it to your local Relay which forwards it to Sentry.

Reviewer: ```suggestion
In sum, this process affects the time it takes to send a request to Sentry, as it is added on top of your servers' response time. To mitigate this and reduce the addition close to zero, you can use a Relay running locally on the same machine or local network that acts as a proxy/agent. As a result, instead of your PHP process sending things to Sentry, the PHP process sends it to your local Relay, which then forwards it to Sentry.
```

> To get started follow the Docs we have about Relay [Getting Started](/product/relay/getting-started/).

Reviewer: ```suggestion
To begin using Relay, review our content for [Getting Started](/product/relay/getting-started/).
```

> Right now we recommend using Relay in `managed` mode (which is the default), follow the instructions in the Relay docs until you manage to send a Test Event via Relay to Sentry.

Reviewer: ```suggestion
Currently, we recommend using Relay in `managed` mode (which is the default). Then follow the instructions in the Relay docs to send a Test Event via Relay to Sentry.
```

Don't forget to update your `DSN` to point to your running Relay instance.

> After this is done, it should dramatically improve the impact on your server.

Reviewer: ```suggestion
After you set up Relay, you should see a dramatic improvement to the impact on your server.
```

---
title: Performance
description: "Learn more about how to configure our Performance integrations to get the best experience out of it."
---

> 

Reviewer 2: We could use an introductory paragraph to break the ice and set the context (good for SEO and for readers) before diving into specific config.

: `traces_sampler`

If you want to sample specific Transaction, use the `traces_sampler` option:

```php
Sentry\init([
    'dsn' => '___PUBLIC_DSN___',
    'traces_sampler' => static function (Sentry\Tracing\SamplingContext $context): float {

>         // This example will discard all Transaction that have health in their name
>         if (false === strpos($context->getTransactionContext()->getName(), 'health')) {
>             return 1.0;
>         }
>         return 0;

Reviewer 2: I think reversing the condition makes it easier to understand what sampling rate applies to what situation better:

```suggestion
        if (false !== strpos($context->getTransactionContext()->getName(), 'health')) {
            // Discard transactions that have 'health' in their name
            return 0.0;
        }
        // Sample rate for all other transactions
        return 1.0;
```

    },
]);
```

> ### Manual Instrumentation

Reviewer 2: Is there no automatic instrumentation for Laravel?

Author: There is; Good catch I will document what we automatically track.

To manually instrument certain regions of your code, you can create a transaction to capture them.

```php
$transactionContext = new TransactionContext();
$transactionContext->setName('External Call');
$transactionContext->setOp('http.caller');
$transaction = \Sentry\startTransaction($transactionContext);

$spanContext = new SpanContext();
$spanContext->setOp('functionX');
$span1 = $transaction->startChild($spanContext);

// Calling functionX
$this->functionX();
$span1->finish();

$transaction->finish();
```

### Retrieving a Transaction

In cases where you want to attach Spans to an already ongoing Transaction you can use `SentrySdk::getCurrentHub()->getTransaction()`. This function will return a `Transaction` in case there is a running Transaction on the scope, otherwise it returns `null`.

> ```javascript

Reviewer 2: ```suggestion
```php
```
